RFS/src/main/java/com/rfs/common/LuceneDocumentsReader.java

```diff
@@ -16,37 +20,64 @@
 import reactor.core.publisher.Flux;
 
 @RequiredArgsConstructor
+@Slf4j
 public class LuceneDocumentsReader {
-    private static final Logger logger = LogManager.getLogger(LuceneDocumentsReader.class);
-    protected final Path luceneFilesBasePath;
+    public static final int NUM_DOCUMENTS_BUFFERED = 1024;
+    protected final Path indexDirectoryPath;
 
-    public Flux<Document> readDocuments(String indexName, int shardId) {
-        Path indexDirectoryPath = luceneFilesBasePath.resolve(indexName).resolve(String.valueOf(shardId));
-
-        return Flux.using(
-            () -> openIndexReader(indexDirectoryPath),
-            reader -> {
-                logger.info(reader.maxDoc() + " documents found in the current Lucene index");
+    public class DocumentOrDone {}
+    @Getter
+    @AllArgsConstructor
+    public class DocumentHolder extends DocumentOrDone {
+        Document value;
+    }
+    public class Done extends DocumentOrDone {}
 
-                return Flux.range(0, reader.maxDoc()) // Extract all the Documents in the IndexReader
-                .handle((i, sink) -> {
-                    Document doc = getDocument(reader, i);
+    public BlockingQueue<DocumentOrDone> packageDocumentsIntoQueue() {
+        var docQueue = new ArrayBlockingQueue<DocumentOrDone>(NUM_DOCUMENTS_BUFFERED, true);
+        new Thread(() -> {
```

Comment:
Lets source the thread from specific threadpool

Reply:
I haven't gotten the pub-sub code working yet.  I'll revisit this later.  It isn't actually critical for this PR since I did what I needed to do with Fluxes after all.